pages/index.tsx

```diff
@@ -0,0 +1,245 @@
+  dt: number;
+  dt_txt:string;
+  main: {
+    temp: number;
+    feels_like: number;
+    temp_min: number;
+    temp_max: number;
+    humidity: number;
+  };
+  weather: {
+    id: number;
+    main: string;
+    description: string;
+    icon: string;
+  }[];
+  clouds: {
+    all: number;
+  };
+  wind: {
+    speed: number;
+    deg: number;
+    gust: number;
+  };
+}
+
+// format in which forecast for 5 days api returns the data for a given location
+interface WeatherData {
+  cod: string;
+  message: number;
+  cnt: number;
+  list: WeatherDetail[];
+  city: {
+    id: number;
+    name: string;
+    coord: {
+      lat: number;
+      lon: number;
+    };
+    country: string;
+    population: number;
+    timezone: number;
+    sunrise: number;
+    sunset: number;
+  };
+}
+
+// props of Linechart
+interface Chart {
+  dates:string,
+  temp:number
+}
+
+//getting icon based on current weather
+export function getIconUrl(code: string): string {
+  return `http://openweathermap.org/img/wn/${code}.png`;
+}
+
+export default function Home() {
+  const API_KEY = process.env.NEXT_PUBLIC_API_KEY;
+  const BASE_URL = process.env.NEXT_PUBLIC_BASE_URL;
+  const [weatherData,setWeatherData] = useState<WeatherDetail>();
+  const [latitude,setLatitude] = useState(0);
+  const [longitude,setLongitude] = useState(0);
+  const [place,setPlace] = useState("Japan");
+  const [forecast,setForecast] = useState<WeatherData>();
+
+  const currentDate = new Date();
+  const year = currentDate.getFullYear();
+  const month = currentDate.getMonth() + 1;
+  const day = currentDate.getDate();
+  const todayDate= `${year}/${month}/${day}`;
+  
+  //page routing
+  const searchParams = useSearchParams();
+  const pathname = usePathname();
+  const { replace } = useRouter();
+  const handleOnChange = async (value:string) => {
+    setPlace(value);
+  };
+
+  useEffect(() => {
+    axios
+        .get(`https://api.openweathermap.org/geo/1.0/direct?q=${place}&limit=5&appid=${API_KEY}`) //getting coordinates based on place selected by user
+        .then((response) => {
+            const data = response.data;
+            setLatitude(data[0].lat);
+            setLongitude(data[0].lon)
+        });
+    axios
+        .get(`${BASE_URL}weather?lat=${latitude}&lon=${longitude}&appid=${API_KEY}&units=metric`) //getting current data
+        .then((response) => {
+            const data = response.data;
+            setWeatherData(data);
+        });
+    axios
+      .get(`${BASE_URL}forecast?lat=${latitude}&lon=${longitude}&appid=${API_KEY}&units=metric`) //getting next 5days data
+      .then((response) => {
+          const data = response.data;
+          setForecast(data);
+      });
+      const params = new URLSearchParams(searchParams); //updating url
+      if (place) {
+        params.set('query', place);
+      } else {
+        params.delete('query');
+      }
+      replace(`${pathname}?${params.toString()}`);
+    
+}, [place,latitude,longitude]); 
```

Comment:
A couple of concerns here:

1. The 2nd and 3rd axios calls seem to be relying on the first, but these are called concurrently. So, the behavior can be unpredictable.
2. I see that the dependencies are `[place,latitude,longitude]` but inside this useEffect there is an updater of lat and long as well (`setLatitude(data[0].lat)`.) Would this cause an infinite loop? Maybe you need only `[place]`

Reply:
1. You are right I have moved the second and third call to another useEffect function now. It should be fine now I guess ?
2. Yes only place would have been enough, but in both cases it was making calls to useEffect twice, which was not good. I think as I have separated them now, it should be calling it only once now ?